File: day-2/python/part-1.py
#!/usr/bin/env python3

position = 0
depth = 0

while True:
    try:
        command = input()
    except EOFError:
        break

    direction, distance = command.split(" ")

    if direction == "forward":
        position += int(distance)
    elif direction == "up":
        depth -= int(distance)
    elif direction == "down":
        depth += int(distance)
    else:
        pass

print(position * depth)

# Answer: 1714680

File: day-2/rust/part-1.rs
#!/usr/bin/env cargo-script

use std::io;

fn main() {
    let mut position = 0;
    let mut depth = 0;

    loop {
        let mut input = String::new();
        match io::stdin().read_line(&mut input) {
            Ok(n) => {
                if n == 0 {
                    break;
                }
                input = input.trim().to_string();
            }
            Err(error) => {
                eprintln!("{}", error);
            }
        }

	let mut iter = input.split_whitespace();

        let direction = iter.next().unwrap();
        let distance = iter.next().unwrap().parse::<i32>().unwrap();

        match direction {
            "forward" => position += distance,
            "down" => depth += distance,
            "up" => depth -= distance,
            _ => (),
        }
    }

    println!("{}", position * depth);
}

// Answer: 1714680
